### PR TITLE
feat: manually handle exchange events - resolve modal

### PR DESCRIPTION
**Pull request by michaelmajgaard**

### CODE REVIEW

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+})
+export class DhResolveModalComponent extends WattTypedModal<{
+  resolve: (comment: string) => Promise<boolean>;
+}> {
+  private readonly formBuilder = inject(FormBuilder);
+
+  busy = false;
+
+  resolveForm = this.formBuilder.group({
+    comment: ['', [Validators.required, Validators.maxLength(1024), Validators.minLength(1)]],
```
Comment (defectiveAi) [inline libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts:75]: nitpick: minLength should be redundant, when required is there.

**Review by FirestarJes — DISMISSED** [review]

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-outgoing-message-drawer.component.ts [hunk 2]
```diff
@@ -183,4 +190,46 @@ export class DhOutgoingMessageDrawerComponent {
...
+      });
+      return false;
+    }
+
+    this.toastService.open({
+      type: 'success',
+      message: translate('eSett.outgoingMessages.drawer.resolveModal.resolvedSuccess'),
+    });
+
+    this.loadOutgoingMessage(this.outgoingMessage.documentId);
```
Comment (defectiveAi) [inline libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-outgoing-message-drawer.component.ts:223]: Will this reload the status in list too?

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts (hunk 1 above)
Comment (michaelmajgaard) [reply libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts:75]: Fair, I remove

**Review by defectiveAi — APPROVED** [review]

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-outgoing-message-drawer.component.ts (hunk 2 above)
Comment (mimse) [reply libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-outgoing-message-drawer.component.ts:225]: You can use refetchQueries on the mutations so you dont need to refetch

https://github.com/Energinet-DataHub/greenforce-frontend/blob/1c0e568d4e8d77735583a55e59147186da4fd91f/libs/dh/profile/feature-profile-modal/src/lib/dh-profile-modal/dh-profile-modal.component.ts#L136

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    WattButtonComponent,
+    WattTextAreaFieldComponent,
+  ],
+})
+export class DhResolveModalComponent extends WattTypedModal<{
+  resolve: (comment: string) => Promise<boolean>;
+}> {
+  private readonly formBuilder = inject(FormBuilder);
+
+  busy = false;
```
Comment (mimse) [inline libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts:72]: Convert this to a signal

**Review by mimse — CHANGES_REQUESTED** [review]
Would it be better to move the mutation to the modal instead. When pass the error state back to the drawer component to show the toast.
Then you dont need to parse in an async function, and you will have access to the loading / error state in the modal component

File: libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts (hunk 3 above)
Comment (michaelmajgaard) [reply libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts:72]: Why?

Comment (michaelmajgaard) [reply libs/dh/esett/feature-outgoing-messages/src/lib/drawer/dh-resolve-modal.component.ts:81]: Moved mutation to modal.

**Review by mimse — APPROVED** [review]

**Pull request merged**